python/python/tests/test_pubsub.py

```diff
@@ -0,0 +1,486 @@
+from __future__ import annotations
+import pytest
+from typing import List, Set, Dict, Tuple, Optional, Any, Awaitable
+import asyncio
+import fnmatch
+
+from tests.conftest import create_client
+
+from glide.async_commands.core import (
+    CoreCommands,
+)
+
+
+MessagesByExactChannel = Dict[str, Dict[str, int]] # channel -> {msg -> cnt}
+MessagesByChannelAndChannel = Dict[Tuple[str, Optional[str]], Dict[str, int]] # (channel, pattern) -> {msg -> cnt}
+SubscriptionsByMode = Dict[CoreCommands.ChannelModes, Set[str]] # subscription_type -> {channel_or_pattern}
+
+
+@pytest.mark.asyncio
+class TestPubSub:
+
+    class PubSubClient:
+        def __init__(self,
+                     subsciptions: SubscriptionsByMode,
+                     sharded_messages_to_publish: MessagesByExactChannel,
+                     unsharded_messages_to_publish: MessagesByExactChannel,
+                     cluster_mode: bool,
+                     request):
+            self.messages_received: MessagesByChannelAndChannel = {}
+            self.messages_to_receive: MessagesByChannelAndChannel = {}
+
+            self.sharded_messages_to_publish: MessagesByExactChannel = sharded_messages_to_publish
+            self.unsharded_messages_to_publish: MessagesByExactChannel = unsharded_messages_to_publish
+
+            self.subsciptions = subsciptions
+
+            self.cluster_mode = cluster_mode
+            self.request = request
+
+            self.redis_client: CoreCommands = None
```

Comment:
CoreCommands? should be TRedisClient

Reply:
why do I need union? - I am only using the CoreCommands methods?

Reply:
done